### PR TITLE
Allow a broker to be removed from the ClusterTopology

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

This PR adds a TopologyChangeOperation for removing a member from the ClusterTopology. A member can leave only if it doesn't replicate any partitions. When the leave operation is applied, the member state is updated to LEFT, but it is not removed from the topology. 

In addition,it adds a lastUpdated timestamp to `MemberState`. This is useful for two reasons
1. Observability. We can track how long a member has been in a particular state. This is especially useful when the topology change operation is not making progress.
2. Might be useful to garbage collect members that are in state LEFT.

I initially used `int64` to encode lastUpdated timestamp. But protobuf best practices recommends to use pre-define type Timestamp. 

## Related issues

related #14353 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Review by lenaschoenburg — APPROVED** [review]
:rocket:

File: topology/src/main/resources/proto/topology.proto [hunk 1]
```diff
@@ -15,8 +17,9 @@ message ClusterTopology {
 
 message MemberState {
   int64 version = 1;
-  State state = 2;
-  map<int32, PartitionState> partitions = 3;
+  google.protobuf.Timestamp lastUpdated = 2;
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply topology/src/main/resources/proto/topology.proto:20]: Right. I will do it in a follow up.

File: topology/src/main/java/io/camunda/zeebe/topology/serializer/ProtoBufSerializer.java [hunk 2]
```diff
@@ -230,6 +246,8 @@ private TopologyChangeOperation decodeOperation(
           topologyChangeOperation.getPartitionLeave().getPartitionId());
     } else if (topologyChangeOperation.hasMemberJoin()) {
       return new MemberJoinOperation(MemberId.from(topologyChangeOperation.getMemberId()));
+    } else if (topologyChangeOperation.hasMemberLeave()) {
```
[the remark this answers is not in the document]
Comment (deepthidevaki) [reply topology/src/main/java/io/camunda/zeebe/topology/serializer/ProtoBufSerializer.java:249]: I think it is ok. Tests should capture them.

**Comment by deepthidevaki** [comment]
bors merge

**Comment by deepthidevaki** [comment]
Flaky test https://github.com/camunda/zeebe/pull/14388 

bors merge

**Pull request merged**